File: ABC/106/B.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <math.h>
#include <numeric>
#include <set>
#include <list>
#include <bitset>
using namespace std;

int main()
{
    int N;
    cin >> N;
    int ans = 0;
    for (int i = 1; i <= N; i += 2)
    {
        int tmp = 0;
        for (int j = 1; j <= i; j++)
        {
            if (!(i % j))
            {
                tmp++;
            }
        }
        if (tmp == 8)
        {
            ans++;
        }
    }
    cout << ans << endl;
    return 0;
}

File: ABC/106/B.py
N = int(input())
if N < 105:
    print(0)
else:
    ans = 1
    for a in range(107, N+1, 2):
        tmp = 2
        for b in range(2, a//2):
            if a % b == 0:
                tmp += 1
        if tmp == 8:
            ans += 1
    print(ans)
